onnxscript/function_libs/torch_aten/ops/nn.py

```diff
@@ -780,30 +779,12 @@ def aten_nll_loss(
 ) -> TFloat:
     """nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor"""
 
-    if reduction == 0:  # "none"
-        result = _aten_nll_loss_onnx(
-            self, target, weight, reduction="none", ignore_index=ignore_index
-        )
-    elif reduction == 1:  # "mean"
-        result = _aten_nll_loss_onnx(
-            self, target, weight, reduction="mean", ignore_index=ignore_index
-        )
-    else:  # "sum"
-        result = _aten_nll_loss_onnx(
-            self, target, weight, reduction="sum", ignore_index=ignore_index
-        )
-
-    return result
-
-
-@torch_op("aten::nll_loss", overload=True)
-def _aten_nll_loss_onnx(
-    self: TFloat,
-    target: TInt,
-    weight: TFloat = None,
-    reduction: str = "none",
-    ignore_index: int = -100,
-) -> TFloat:
+    if reduction == 0:
```

Comment:
I don’t think we need to make it trace only ? That would be the whole point of combining.

Reply:
if remove trace_only, we also need to change below file:
[param_manipulation.py](https://github.com/xiaowuhu/onnx-script/blob/430abeebef77d451fd2cc5789f832ce396fee385/onnxscript/_internal/param_manipulation.py#L73)
because the input weight is not passed into the function.